File: testdata/java/Handlers.java
// Decision 11 default-interface-method (extracted as Function), Decision 6
// records-as-Class, sealed interface (permits ignored), static-import.

package handlers;

import static java.lang.Math.max;

public class Handlers {

    public interface IGreeter {
        // Abstract — produces NO symbol (forward-declaration rule).
        void required();

        // Default interface method — Decision 11 extracts as Function
        // (no parent), matching Rust's trait-default-method rule.
        default void greet() {
            required();
        }

        // Static interface method (Java 8+) — also extracts as Function.
        static String banner() {
            return "hi";
        }
    }

    // Sealed interface — Decision 6: `permits` clause is ignored and
    // extracts as ordinary Interface.
    public sealed interface Shape permits Circle, Square { }

    public static final class Circle implements Shape { }
    public static final class Square implements Shape { }

    // record User — Decision 6: extracts as Class. Method inside the
    // record body extracts as Method with parent = `User`.
    public record User(String name) {
        public String display() {
            return name;
        }
    }

    public static class Hub {
        public void dispatch(java.util.List<String> items) {
            for (String item : items) {
                process(item);
            }
            max(1, 2);
        }

        public static void process(String item) { }
    }
}

File: testdata/csharp/Handlers.cs
// Default interface methods (Decision 11), records (Decision 6 analog),
// extension methods (Decision 5), using static + alias + global.

using static System.Math;
using StrList = System.Collections.Generic.List<string>;
global using System.Linq;

namespace Handlers
{
    public interface IGreeter
    {
        // Abstract — produces NO symbol (forward-declaration rule).
        void Required();

        // Default interface method (C# 8+) — extracts as Function per
        // Decision 11's C# follow-up.
        void Greet() { Required(); }
    }

    public record User(string Name)
    {
        public string Display() { return Name; }
    }

    public static class StringExt
    {
        // Extension method (Decision 5): parent is the syntactic
        // enclosing static class `StringExt`, NOT `string`.
        public static int CountWords(this string s)
        {
            return Abs(s.Length);
        }
    }

    public class Hub
    {
        public void Dispatch(StrList items)
        {
            foreach (var item in items)
            {
                Process(item);
            }
        }

        public static void Process(string item) { }
    }
}
